src/lib/components/Searchbar/SearchBar.jsx

```diff
@@ -0,0 +1,160 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { Actions, prepareLayer, lngLat } from 'gisida';
+import './SearchBar.scss'
+
+/**
+ * To activate the search functionality add "searchBar":true and 
+ * To Change search bar color add "searchBarColor": <desired color> 
+ * on client site-config.json APP object
+*/
+
+
+const mapStateToProps = (state, ownProps) => {
+    const { APP, LOC } = state;
+    const MAP = state[ownProps.mapId] || { layers: {} };
+    return {
+      APP,
+      LOC,
+      appColor: APP.appColor,
+      searchBarColor: APP.searchBarColor,
+      preparedLayers: MAP.layers,
+      ...ownProps
+    };
+  };
+
+class SearchBar extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      inputText: '',
+      selectedLayerId: null,
+    }
+    this.handleCancel = this.handleCancel.bind(this);
+    this.handleSearchInput = this.handleSearchInput.bind(this);
+  }
+
+  componentDidUpdate() {
+    const { selectedLayerId } = this.state;
+    const { preparedLayers, openCategoryForSharedLayers } = this.props;
+
+    // open menu when layer is loaded visibility is turned on
+    if (selectedLayerId && preparedLayers[selectedLayerId].visible) {
+      const toOpenLayer = {
+        id: selectedLayerId,
+        isCatOpen: false,
+      };
+      openCategoryForSharedLayers([toOpenLayer]);
+      this.setState({selectedLayerId: null});
+    }
```

Comment:
Now this is a good example of setting state in componentDidUpdate in a way that will not cause multiple / unnecessary loops! 👏